pkg/experiment/query_backend/block_reader.go

```diff
@@ -132,3 +155,18 @@ func validateRequest(req *queryv1.InvokeRequest) (*request, error) {
 	}
 	return &r, nil
 }
+
+func filterInvalidDatasets(b *metastorev1.BlockMeta, tenantMap map[string]struct{}) ([]*metastorev1.Dataset, error) {
+	errs := multierror.New()
+	datasets := make([]*metastorev1.Dataset, 0, len(b.Datasets))
```

Comment:
We probably could get rid of that allocation, esp. since we would not expect this to ever be hit (the normal case is our dataset tenant filtering works).